optuna/samplers/_gp/_sampler.py

```diff
@@ -0,0 +1,178 @@
+from __future__ import annotations
+
+from typing import Any
+from typing import Callable
+from typing import Sequence
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+import optuna
+from optuna._experimental import experimental_class
+from optuna._imports import _LazyImport
+from optuna.distributions import BaseDistribution
+from optuna.samplers._base import BaseSampler
+from optuna.study import Study
+from optuna.study import StudyDirection
+from optuna.trial import FrozenTrial
+from optuna.trial import TrialState
+
+
+if TYPE_CHECKING:
+    import torch
+
+    import optuna._gp._acqf as _acqf
+    import optuna._gp._gp as _gp
+    import optuna._gp._optim as _optim
+    import optuna._gp._search_space as _search_space
+else:
+    torch = _LazyImport("torch")
+    _search_space = _LazyImport("optuna._gp._search_space")
+    _gp = _LazyImport("optuna._gp._gp")
+    _optim = _LazyImport("optuna._gp._optim")
+    _acqf = _LazyImport("optuna._gp._acqf")
+
+
+def log_prior(kernel_params: "_gp.KernelParams") -> "torch.Tensor":
+    # Log of prior distribution of kernel parameters.
+
+    def gamma_log_prior(x: "torch.Tensor", concentration: float, rate: float) -> "torch.Tensor":
+        return (concentration - 1) * torch.log(x) - rate * x
+
+    return (
+        gamma_log_prior(kernel_params.inv_sq_lengthscales, 2, 0.5).sum()
+        + gamma_log_prior(kernel_params.kernel_scale, 2, 1)
+        + gamma_log_prior(kernel_params.noise, 1.1, 20)
+    )
```

Comment:
Do you have any reasons why you picked these values?

```suggestion
    return (
        gamma_log_prior(kernel_params.inv_sq_lengthscales, 2, 0.5).sum()
        + gamma_log_prior(kernel_params.kernel_scale, 2, 1)
        + gamma_log_prior(kernel_params.noise, 1.1, 20)
    )
```

Reply:
No, I just visualized these priors and picked ones that looked sensible.
I checked the BoTorch default priors, but BoTorch defaults didn't look very sensible. (Possibly BoTorch default doesn't assume the inputs (X and Y) are normalized/standardized, or the implementer possibly mistook GPyTorch's definition of `Gamma` distribution I think.)

Reply:
Also since we are using different prior distribution than BoTorch (we use gamma prior on `inv_sq_lengthscales` while BoTorch uses gamma prior on `lengthscales`), we cannot use the same parameters anyway. I feel using gamma prior on `lengthscales` would place too much prior on very small lengthscales, which is very bad (I confirm that in many cases `BoTorch` does indeed falls back to very small lengthscales. Small lengthscales mean no information, so that would render optimization useless.)

Reply:
Thank you for the clarification!
Then let's just make a note about it:)